content/pytorch/concepts/using-functional-apis/using-functional-api.md

```diff
@@ -0,0 +1,79 @@
+---
+Title: 'Using Functional APIs'
+Description: 'Functional APIs in PyTorch provide a flexible way to define and manipulate neural networks using functions rather than object-oriented classes.'
+Subjects:
+  - 'Machine Learning'
+  - 'Data Science' 
+Tags:
+  - 'PyTorch'
+  - 'Deep Learning'
+  - 'Neural Networks'
+CatalogContent:
+  - 'learn-python'
+  - 'paths/data-science'
+---
+
+**Functional APIs** in PyTorch provide a flexible and powerful way to define and manipulate neural networks. Unlike the `torch.nn.Module` class, which uses an object-oriented approach, functional APIs allow you to define models using functions. This can be particularly useful for creating complex models, experimenting with new architectures, or when you need more control over the forward pass.
+
+## Key Concepts
+
+### Functional Layers
+
+Functional layers are stateless and do not store parameters. Instead, parameters are passed explicitly. This can be useful for creating custom layers or when you need to reuse the same layer with different parameters.
+
+Example:
+
+```python
+import torch
+import torch.nn.functional as F
+
+# Define input tensor
+x = torch.randn(10, 3)
+
+# Define weights and bias
+weight = torch.randn(5, 3, requires_grad=True)
+bias = torch.randn(5, requires_grad=True)
+
+# Apply linear transformation
+output = F.linear(x, weight, bias)
+```
+
+### Custom Loss Functions
+
+Using functional APIs, you can easily define custom loss functions. This is useful when the predefined loss functions in PyTorch do not meet your requirements.
+
+Example:
+```python
+def custom_loss(output, target):
+    loss = torch.mean((output - target) ** 2)
+    return loss
+
+# Example usage
+output = torch.randn(10, 5)
+target = torch.randn(10, 5)
+loss = custom_loss(output, target)
```

Comment:
Add a print statement to print the loss and also include the output here.

```suggestion
loss = custom_loss(output, target)
print("Custom MSE Loss:", loss)
```

The output might look something like this (the actual number will change due to randomness):

```shell
Custom MSE Loss: tensor(1.2135)
```